src/content/docs/en/reference/loader-reference.mdx

```diff
@@ -3,6 +3,450 @@ title: Astro Loader API
+      }
+      // ...
+    },
+  };
+}
+```
+
+### `DataStore`
+
+The data store is a loader's interface to the content collection data. It is a key value store, scoped to the collection, so a loader can only access the data for its own collection.
+
+#### `get`
+
+<p>
+
+**Type**: <code>(key: string) => <a href="#dataentry">DataEntry</a> | undefined</code>
+</p>
+
+Get an entry from the store by its ID. Returns `undefined` if the entry does not exist. 
+
+```ts
+const existingEntry = store.get("my-entry");
+```
+
+The returned object is a [`DataEntry`](#dataentry) object.
+
+#### `set`
+
+<p>
+
+**Type**: <code>(entry: <a href="#dataentry">DataEntry</a>) => boolean</code>
+</p>
+
+Adds an entry to the store, returning `true` if the entry was set. If `digest` was passed and matched an existing entry with the same ID, it will return `false` and not update the entry. See [`DataEntry`](#dataentry) for more information on the fields to pass. The entry should have been parsed using `parseData` before being stored. As well as applying any validations and transforms in the schema, this will also ensure that any images are correctly processed.
+
+```ts title=loader.ts {7-14}
+    for (const item of feed.items) {
+      const data = await parseData({
+        id: item.guid,
+        data: item,
+      });
+      const digest = generateDigest(data);
+      store.set({
+        id,
+        data,
+        rendered: {
+          html: data.description ?? "",
+        },
+        digest,
+      });
+    }
+```
+
+#### `entries`
+
+<p>
+
+**Type**: `() => Array<[id: string, DataEntry]>`
+</p>
+
+Get all entries in the collection as an array of key-value pairs.
+
+#### `keys`
+**Type**: `() => Array<string>`
+
+Get all the keys of the entries in the collection.
+
+#### `values`
+
+<p>
+
+**Type**: `() => Array<DataEntry>`
+</p>
+
+Get all entries in the collection as an array.
+
+#### `delete`
+
+<p>
+
+**Type**: `(key: string) => void`
+</p>
+
+Delete an entry from the store by its ID.
+
+#### `clear`
+
+<p>
+
+**Type**: `() => void`
+</p>
+
+Clear all entries from the collection.
+
+#### `has`
+
+<p>
+
+**Type**: `(key: string) => boolean`
+</p>
+
+Check if an entry exists in the store by its ID.
+
+### `DataEntry`
+
+This is the type of the object that is stored in the data store. It has the following properties:
+
+#### `id`
+
+<p>
+
+**Type**: `string`
+</p>
+
+An identifier for the entry, which must be unique within the collection. This is used to look up the entry in the store, and is the key used with `getEntry` for that collection. 
+
+#### `data`
+
+<p>
+
+**Type**: `Record<string, unknown>`
+</p>
+
+The actual data for the entry. When a user accesses the collection, this will have TypeScript types generated according to the collection schema. It the responsibility of the loader to use [`parseData`](#parsedata) to validate and parse the data before storing it in the data store: no validation is done when getting or setting the data.
```

Comment:
```suggestion
The actual data for the entry. When a user accesses the collection, this will have TypeScript types generated according to the collection schema.

It the loader's responsibility to use [`parseData`](#parsedata) to validate and parse the data before storing it in the data store: no validation is done when getting or setting the data.
```
Just separating paragraphs!